mmdet/ops/nms/nms_wrapper.py

```diff
@@ -1,218 +1,22 @@
-    elif isinstance(dets, np.ndarray):
-        is_numpy = True
-        device = 'cpu' if device_id is None else f'cuda:{device_id}'
-        dets_th = torch.from_numpy(dets).to(device)
-    else:
-        raise TypeError('dets must be either a Tensor or numpy array, '
-                        f'but got {type(dets)}')
-
-    if max_num < 0:
-        max_num = int(dets_th.size(0))
-
-    inds = nms_core(dets_th, iou_threshold, score_thr, max_num)
-
-    if is_numpy:
-        inds = inds.cpu().numpy()
-    return dets[inds, :], inds
-
-
-@py_symbolic()
-def nms_core(dets, iou_thr, score_thr, max_num):
-    if is_in_onnx_export():
-        with no_nncf_trace():
-            valid_dets_mask = dets[:, 4] > score_thr
-        dets = dets[valid_dets_mask]
-
-    if dets.shape[0] == 0:
-        inds = dets.new_zeros(0, dtype=torch.long)
-    else:
-        inds = nms_ext.nms(dets, iou_thr)
-
-    if is_in_onnx_export():
-        inds = inds[:max_num]
-
-    return inds
-
-
-def soft_nms(dets, iou_threshold, method='linear', sigma=0.5, min_score=1e-3):
-    """Dispatch to only CPU Soft NMS implementations.
-
-    The input can be either a torch tensor or numpy array.
-    The returned type will always be the same as inputs.
-
-    Arguments:
-        dets (torch.Tensor or np.ndarray): bboxes with scores.
-        iou_threshold (float): IoU threshold for Soft NMS.
-        method (str): either 'linear' or 'gaussian'
-        sigma (float): hyperparameter for gaussian method
-        min_score (float): score filter threshold
-
-    Returns:
-        tuple: new det bboxes and indice, which is always the same
-        data type as the input.
-
-    Example:
-        >>> dets = np.array([[4., 3., 5., 3., 0.9],
-        >>>                  [4., 3., 5., 4., 0.9],
-        >>>                  [3., 1., 3., 1., 0.5],
-        >>>                  [3., 1., 3., 1., 0.5],
-        >>>                  [3., 1., 3., 1., 0.4],
-        >>>                  [3., 1., 3., 1., 0.0]], dtype=np.float32)
-        >>> iou_thr = 0.6
-        >>> new_dets, inds = soft_nms(dets, iou_thr, sigma=0.5)
-        >>> assert len(inds) == len(new_dets) == 5
+    This function updates 'score_threshold' and 'max_num' in the next 'NMSop' call. 
+    It should be replaces with 'mmcv::batched_nms' after adding support for 'score_threshold' and 'max_num' in MMCV.
     """
-    # convert dets (tensor or numpy array) to tensor
-    if isinstance(dets, torch.Tensor):
-        is_tensor = True
-        dets_t = dets.detach().cpu()
-    elif isinstance(dets, np.ndarray):
-        is_tensor = False
-        dets_t = torch.from_numpy(dets)
+    if torch.onnx.is_in_onnx_export():
+        score_thr = nms_cfg.pop('score_thr') if 'score_thr' in nms_cfg.keys() else 0.0
+        max_num = nms_cfg.pop('max_num') if 'max_num' in nms_cfg.keys() else int(bboxes.size(0))
+    
+        from ...utils.deployment.symbolic import set_extra_args_for_NMSop
+        set_extra_args_for_NMSop(score_thr, max_num)
```

Comment:
The whole story looks somewhat overcomplicated...
IMO, patching methods of some 3rd-party package is reasonable as long as no interface changes introduced. Here this is not the case. Taking into account potential need of keeping customized code for `batched_nms` (to have NNCF-related stuff), I'd consider more explicit approach: redefine `batched_nms` and `nms` functions adding extra required arguments.